src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -2754,40 +2754,71 @@ QList< QgsProviderSublayerDetails > QgsOgrProviderUtils::querySubLayerList( int
       }
     }
 
-    // List TIN and PolyhedralSurface as Polygon
-    if ( fCount.contains( wkbTIN ) )
+    QMap<Qgis::GeometryType, size_t> baseTypeCount;
+    baseTypeCount[Qgis::GeometryType::Point] = 0;
+    baseTypeCount[Qgis::GeometryType::Line] = 0;
+    baseTypeCount[Qgis::GeometryType::Polygon] = 0;
+
+    OGRwkbGeometryType polyBaseType { wkbPolygon };
+    OGRwkbGeometryType lineBaseType { wkbLineString };
+    OGRwkbGeometryType pointBaseType { wkbPoint };
+
+    // Last type in the list is the winner
+    const static QList<OGRwkbGeometryType> pointHyerarchy { wkbPoint, wkbMultiPoint };
```

Comment:
typo here and in the 2 other variables:
```suggestion
    const static QList<OGRwkbGeometryType> pointHierarchy { wkbPoint, wkbMultiPoint };
```